rfc/plugins.md

```diff
@@ -0,0 +1,248 @@
+
+An extension looks like the following:
+
+```kotlin
+// Definition of the PluginStage extension point.
+// The annotation defined here also informs the Spinnaker
+// plugin framwework which runtimes are acceptable. This 
+// annotation would additionally be used for other metadata
+// Spinnaker needs to know atop whatever PF4J offers out of
+// the box.
+@SpinnakerExtension(drivers = [
+  ExtensionDriver.LOCAL, 
+	ExtensionDriver.REMOTE
+])
+class PipelineStageExtension : ExtensionPoint {
+	// ...
+}
+```
+
+It is highly recommended to read the documentation of PF4J to fully understand the scope of this proposal.
+
+### Front50 as the Source of Truth
+Operators will need the ability to understand, at a glance, what plugins are installed and enabled or disabled, their versions, as well as any other metadata about a plugin. Whether plugins are implemented as in-process or not, they Front50 should be used to track and expose a single source of truth of what plugins are installed in any particular service, as well as which ones are enabled or disabled. 
+
+The Front50 plugin registry will also be exposed through Gate, and can be consumed by Deck to load the correct plugins on application startup.
+
+Plugins can also be enabled or disabled at runtime without redeploying a service. This flag is sourced from Front50, which all services will be listening for change events from. Whether this change event is published or polled for initially is unknown.
+
+Halyard configs can be used to inform services of their _initial state_. When a service starts up, it will notify Front50 of what in-process plugins are installed, however Front50 will be responsible for determining the enabled state.
+
+Using Front50 as a source of truth will require extension of PF4J.
+
+```proto
+syntax = "proto3";
+
+// PluginManifest is used during registration of a plugin
+message PluginManifest {
+	// The name of the plugin, which is used as a fully 
+	// qualified identifier, including a namespace and 
+	// plugin name, ex: "netflix/fast-properties"
+	string name = 1;
+
+	// A short description of the plugin
+	// ex: "Provides rollout safety of runtime application 
+	// config via pipelines"
+	string description = 3;
+
+	// The name of the plugin author, ex: "Netflix"
+	string provider_name = 4;
+
+	// The semver plugin version, ex: "1.0.0"
+	string versionn = 5;
+
+	// A list of plugin dependencies. Initially this will be
+	// informational only. ex: "netflix/somedep>=1.x"
+	repeated string dependencies = 6;
+
+	// required_versions is a repeated string of service version	// requirements. e.g. "clouddriver >= 5.33.0"
+	repeated string required_versions = 7;
+
+	// A list of extension points that the plugin implements.
+	// e.g. "pipelineStage"
+	repeated string provided_capabilities = 8;
```

Comment:
I think the capabilities be `enum`s

Reply:
I would rather not. Making it an enum would couple the proto to the services more than I'd prefer, and would prevent `-nflx` style extensions from providing plugins of their own without modifying the OSS proto.

Reply:
hm... that's fair, thanks for the explanation.
I guess we could have a "validation" mechanism of sorts for developers before they publish, if we wanted to